File: todoFullstack/src/main/java/com/service/TodoService.java
package com.service;

import com.domain.Todo;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface TodoService {
    public Todo saveTask(Todo todo);
    public List<Todo> getAll();
    public ResponseEntity<Todo> getTaskById(Long id);
    public ResponseEntity<Todo> updateTask(Long id, Todo todo);
    public ResponseEntity<HttpStatus> delete(Long id);
}

File: service/todoService.js
import axios from 'axios';
const getRestApi = "http://localhost:8084/api/v1/getAll"
const saveRestApi = "http://localhost:8084/api/v1/add"
const getRestApiById = "http://localhost:8084/api/v1/getTask"

class TodoService{
    getAll(){
        return axios.get(getRestApi);
    }
    add(todo){
        return axios.post(saveRestApi, todo);
    }
    getTaskById(id){
        return axios.get(getRestApiById + '/' + id);

    }
    updateTask(id, todo){
        return axios.put(getRestApiById + '/'+ id, todo);
    }
    deleteTask(id){
        return axios.delete(getRestApiById + '/' + id);
    }

}
export default new TodoService();